File: RadarHub.Back/RadarHub.API/Controllers/UnidadeController.cs
using Microsoft.AspNetCore.Mvc;
using RadarHub.Dominio.Entidades;
using RadarHub.Dominio.Servicos;
using RSK.API.Controllers;
using RSK.Dominio.Interfaces;
using RSK.Dominio.Notificacoes.Interfaces;
using RSK.Dominio.Servicos;

namespace RadarHub.API.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class UnidadeController : ApiConsultaControllerBase<Unidade>
    {
        private readonly UnidadeServico _unidadeServico;
        private readonly INotificador _mensagens;

        public UnidadeController(IServicoConsultaBase<Unidade> servicoConsulta, INotificador mensagens, UnidadeServico unidadeServico) : base(servicoConsulta)
        {
            _unidadeServico = unidadeServico;
            _mensagens = mensagens;
        }


        [HttpPost("Importar")]
        public async Task<IActionResult> Importar()
        {
            var sucesso = await _unidadeServico.ImportarAsync();
            var resultado = _mensagens.ObterMensagens();

            if (!sucesso)
            {
                return BadRequest(resultado);
            }

            return Ok(resultado);
        }
    }
}

File: RadarHub.Web/src/servicos/UnidadeController.ts
import { Unidade } from "@/DTO/Unidade";
import { ConsultaServicoBase } from "./ConsultaServicoBase";
import http from "./http-comon";

class UnidadeController extends ConsultaServicoBase<Unidade> {
    constructor() {
        super("Unidade")
    }

    public async importar() {
        return http.post(`/unidade/importar`);
    }
}

const UnidadeServico = new UnidadeController();

export default UnidadeServico;